src/lib/validation.ts

```diff
@@ -0,0 +1,35 @@
+import { CreateUrlError } from '$lib/errors';
+
+const allowedProtocols = ['http:', 'https:'];
+const shortUrlRegexp = /^[a-zA-Z0-9-]+$/;
+
+export function validateCreateUrlFormData(formData: FormData) {
+	const url = formData.get('url');
+	const shortUrl = formData.get('shortUrl');
+
+	if (typeof url !== 'string') {
+		throw new CreateUrlError('url', 'URL must be a string');
+	}
+
+	if (typeof shortUrl !== 'string') {
+		throw new CreateUrlError('shortUrl', 'Short URL must be a string');
+	}
+
+	let parsedUrl: URL;
+
+	try {
+		parsedUrl = new URL(url);
+	} catch (_) {
+		throw new CreateUrlError('url', 'Invalid URL');
+	}
+
+	if (!allowedProtocols.includes(parsedUrl.protocol)) {
+		throw new CreateUrlError('url', 'Only HTTP and HTTPS URLs are allowed');
+	}
```

Comment:
I read about URL.canParse method, but i need also check protocol of the actual url and canParse does not allowe this, so the solution was bettwen native api and regexp i choose first approach